spread/qemu.go

```diff
@@ -97,30 +103,65 @@ func systemPath(system *System) string {
 	return os.ExpandEnv("$HOME/.spread/qemu/" + system.Image + ".img")
 }
 
-func (p *qemuProvider) Allocate(ctx context.Context, system *System) (Server, error) {
-	// FIXME Find an available port more reliably.
-	port := 59301 + rand.Intn(99)
-
-	path := systemPath(system)
-	if info, err := os.Stat(path); err != nil || info.IsDir() {
-		return nil, &FatalError{fmt.Errorf("cannot find qemu image at %s", path)}
+func ovmfPath() string {
+	// this will work on ubuntu/debian/fedora
+	// allow packaging to override default
+	if _, err := os.Readlink(optionalOvmfPackageSymlink); err == nil {
+		return optionalOvmfPackageSymlink
 	}
-
-	mem := 1500
-	if p.backend.Memory > 0 {
-		mem = int(p.backend.Memory / mb)
+	// can be used by e.g. the snap
+	if p := os.Getenv("SPREAD_QEMU_OVMF_PATH"); p != "" {
```

Comment:
This needs to be renamed to something like (strawman) SPREAD_QEMU_FALLBACK_BIOS_PATH that looks at 
$SPREAD_QEMU_FALLBACK_BIOS_PATH/uefi.img